src/expr.cpp

```diff
@@ -7708,6 +7708,63 @@ std::pair<llvm::Constant *, bool> SizeOfExpr::GetConstant(const Type *rtype) con
                                              isNotValidForMultiTargetGlobal);
 }
 
+///////////////////////////////////////////////////////////////////////////
+// AllocaExpr
+
+AllocaExpr::AllocaExpr(Expr *e, SourcePos p) : Expr(p, AllocaExprID), expr(e) {}
+
+llvm::Value *AllocaExpr::GetValue(FunctionEmitContext *ctx) const {
+    ctx->SetDebugPos(pos);
+    if (expr == NULL)
+        return NULL;
+    llvm::Value *llvmValue = expr->GetValue(ctx);
+    if (llvmValue == NULL)
+        return NULL;
+    llvm::Value *resultPtr = ctx->AllocaInst((LLVMTypes::VoidPointerType)->getElementType(), llvmValue, "allocaExpr",
+                                             16, false); // 16 byte stack alignment.
+    return resultPtr;
+}
+
+const Type *AllocaExpr::GetType() const { return PointerType::Void; }
+
+void AllocaExpr::Print() const {
+    printf("AllocaExpr (");
+    if (expr != NULL)
+        expr->Print();
+    const Type *t = expr ? expr->GetType() : NULL;
+    if (t != NULL)
+        printf(" [type %s]", t->GetString().c_str());
+    printf(")");
+    pos.Print();
+}
+
+Expr *AllocaExpr::TypeCheck() {
+    if (expr == NULL) {
+        Error(pos, "\"allocate()\" cannot have a \"NULL\" argument.");
```

Comment:
It should be `alloca()` in the error message, not `allocate()`

Reply:
Done